tests/testDegenerateTemplate.py

```diff
@@ -0,0 +1,145 @@
+#
+from __future__ import print_function
+import unittest
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.detection as afwDet
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import lsst.pex.logging as pexLogging
+from lsst.meas.deblender.baseline import deblend
+import lsst.meas.algorithms as measAlg
+
+root = pexLogging.Log.getDefaultLog()
+root.setThreshold(pexLogging.Log.DEBUG)
+
+# Quiet some of the more chatty loggers
+pexLogging.Log(root, 'lsst.meas.deblender.symmetrizeFootprint',
+               pexLogging.Log.INFO)
+pexLogging.Log(root, 'lsst.meas.deblender.symmetricFootprint',
+               pexLogging.Log.INFO)
+pexLogging.Log(root, 'lsst.meas.deblender.getSignificantEdgePixels',
+               pexLogging.Log.INFO)
+pexLogging.Log(root, 'afw.Mask',
+               pexLogging.Log.INFO)
+
+
+def imExt(img):
+    bbox = img.getBBox()
+    return [bbox.getMinX(), bbox.getMaxX(),
+            bbox.getMinY(), bbox.getMaxY()]
+
+
+def doubleGaussianPsf(W, H, fwhm1, fwhm2, a2):
+    return measAlg.DoubleGaussianPsf(W, H, fwhm1, fwhm2, a2)
+
+
+def gaussianPsf(W, H, fwhm):
+    return measAlg.DoubleGaussianPsf(W, H, fwhm)
+
+
+class DegenerateTemplateTestCase(lsst.utils.tests.TestCase):
+
+    def test1(self):
+        '''
+        A simple example: three overlapping blobs (detected as 1
+        footprint with three peaks).  Additional peaks are added near
+        the blob peaks that should be identified as degenerate.
+        '''
+        H, W = 100, 100
+
+        fpbb = afwGeom.Box2I(afwGeom.Point2I(0, 0),
+                             afwGeom.Point2I(W-1, H-1))
+
+        afwimg = afwImage.MaskedImageF(fpbb)
+        imgbb = afwimg.getBBox()
+        img = afwimg.getImage().getArray()
+
+        var = afwimg.getVariance().getArray()
+        var[:, :] = 1.
+
+        blob_fwhm = 10.
+        blob_psf = doubleGaussianPsf(99, 99, blob_fwhm, 2.*blob_fwhm, 0.03)
+
+        fakepsf_fwhm = 3.
+        fakepsf = gaussianPsf(11, 11, fakepsf_fwhm)
+
+        blobimgs = []
+        x = 75.
+        XY = [(x, 35.), (x, 65.), (50., 50.)]
+        flux = 1e6
+        for x, y in XY:
+            bim = blob_psf.computeImage(afwGeom.Point2D(x, y))
+            bbb = bim.getBBox()
+            bbb.clip(imgbb)
+
+            bim = bim.Factory(bim, bbb)
+            bim2 = bim.getArray()
+
+            blobimg = np.zeros_like(img)
+            blobimg[bbb.getMinY():bbb.getMaxY()+1,
+                    bbb.getMinX():bbb.getMaxX()+1] += flux * bim2
+            blobimgs.append(blobimg)
+
+            img[bbb.getMinY():bbb.getMaxY()+1,
+                bbb.getMinX():bbb.getMaxX()+1] += flux * bim2
+
+        # Run the detection code to get a ~ realistic footprint
+        thresh = afwDet.createThreshold(5., 'value', True)
+        fpSet = afwDet.FootprintSet(afwimg, thresh, 'DETECTED', 1)
+        fps = fpSet.getFootprints()
+        print('found', len(fps), 'footprints')
+
```

Comment:
I think this should be quite robust because I'm inserting very bright objects that should all overlap at the 5 sigma level.  I can't see how this could be split into more than one object.